pkg/util/wait/wait.go

```diff
@@ -0,0 +1,22 @@
+package wait
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// PollImmediateWithContext will poll until a stop condition is met
+func PollImmediateWithContext(interval, timeout time.Duration, condition wait.ConditionFunc, stopCh <-chan struct{}) error {
```

Comment:
don't bother with this, just use wait.PollUntil.

Reply:
I think we need timeout as In cases where the installer is busted (and I have seen cases like this, in example "release payload is bad, and master nodes are in "creating" state forever because bootstrap node can't bootstrap when we need a timeout to kill these and return an error. Otherwise, arm deployment is a success. And this basically unifies behaviour for different phases.

Reply:
I agree with you, but PollUntil has a timeout?

Reply:
No,it does not.
```
func Poll(interval, timeout time.Duration, condition ConditionFunc) error
func PollImmediate(interval, timeout time.Duration, condition ConditionFunc) error
```
These are 2 with timeouts. And they dont have context. Added different wrapper to enable timeout with context

Reply:
How about this:
```
	timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Minute)
	defer cancel()
	err = wait.PollUntil(10*time.Second, func() (bool, error) {
		// do stuff
	}, timeoutCtx.Done())
```